File: src/backend/reservation/src/main/java/org/taonaw/reservation/domain/model/equipment/EquipmentStocks.java
package org.taonaw.reservation.domain.model.equipment;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import org.taonaw.reservation.domain.shared.Assertion;

@EqualsAndHashCode
@AllArgsConstructor
public class EquipmentStocks {
    private final int value;

    public boolean isOutOfStocks(int quantity) {
        return quantity > value;
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/equipment/EquipmentStocks.kt
package org.taonaw.facility.domain.model.equipment

data class EquipmentStocks(val value: Int) {

    companion object {
        const val MIN = 0
        const val MAX = 99
    }

    init {
        require(value in MIN..MAX) {
            "機材在庫数は $MIN 個以上 $MAX 個以下でなければなりません。"
        }
    }
}